lib/gitlab_api/base.rb

```diff
@@ -0,0 +1,46 @@
+module GitlabApi
+  class Base
+    attr_accessor :response
+
+    def self.perform(*attrs)
+      new(*attrs).perform
+    end
+
+    def connection
+      @connection ||= Faraday.new(url: gitlab_api_url) do |faraday|
+        faraday.headers['Content-Type'] = 'application/json'
+        faraday.headers['Accept'] = 'application/json'
+        faraday.headers['Private-Token'] = gitlab_private_token
+        faraday.adapter Faraday.default_adapter
+      end
+    end
+
+    def get(path)
+      self.response = connection.get(path)
```

Comment:
not sure about this approach, you are adding self as return, I think in order to make a [fluent interface](https://martinfowler.com/bliki/FluentInterface.html)
and just seting the value to response, why?

Reply:
I gonna need another pair of eyes here @talyssonoc pls

Reply:
@Kutomore is it because we're implementing a fluent interface here? Can you give us an example of code that would use this one?